src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -289,13 +320,21 @@ private AssemblyReferenceHandle GetAssemblyReference(Assembly assembly)
             return handle;
         }
 
-        private GenericParameterHandle AddGenericTypeParameter(TypeDefinitionHandle typeHandle, GenericTypeParameterBuilderImpl gParam) =>
-            _metadataBuilder.AddGenericParameter(
-                parent: typeHandle,
+        private void AddGenericTypeParametersAndConstraintsCustomAttributes(EntityHandle parentHandle, GenericTypeParameterBuilderImpl gParam)
+        {
+            GenericParameterHandle handle = _metadataBuilder.AddGenericParameter(
+                parent: parentHandle,
                 attributes: gParam.GenericParameterAttributes,
                 name: _metadataBuilder.GetOrAddString(gParam.Name),
                 index: gParam.GenericParameterPosition);
 
+            WriteCustomAttributes(gParam._customAttributes, handle);
+            foreach (Type constraint in gParam.GetGenericParameterConstraints())
+            {
+                _metadataBuilder.AddGenericParameterConstraint(handle, GetTypeHandle(constraint));
```

Comment:
The ECMA spec says that `GenericParamConstraint` table needs to be sorted by the owner too. Does this need to go through the same sorting process as GenericParams?

Reply:
Doc says `Constraints must be added in the same order as the corresponding generic parameters.` so adding it for the corresponding generic params is looks correct to me

Reply:
You are right.

Reply:
Added test case for that